### PR TITLE
ISSUE-7986: The topics for the "Company Role" mutations and "isCompanyRoleNameAvailable" query have been created

### LINKED ISSUE #7986: GraphQL: Add company role mutations
# Community engineering issue

Type of Doc:

-  [x] New topic
-  [ ] Topic Update, Devdoc link:
-  [ ] Other

Magento 2 GitHub Issue: https://github.com/magento/magento2/issues/29094

Magento 2 GitHub PR: https://github.com/magento/partners-magento2b2b/pull/203

Magento 2 version:

-  [x] 2.4.2

Notifications of Changes:

-  [ ] Changes to Admin interface?
-  [ ] New software requirements?
-  [x] New REST endpoints or updated GraphQL query syntax?
-  [ ] New command line option?

## Content

Add documentation for the following B2B company role mutations

createCompanyRole
updateCompanyRole
deleteCompanyRole

Also add the documentation for the checkCompanyRoleName query

### PR DESCRIPTION
**Pull request by andrewbess**
## Purpose of this pull request

This pull request (PR) add new topics for "Company Role" mutations and "isCompanyRoleNameAvailable" query for new 2.4.2 features.

fixes #7986 

### Fixed Issues (if relevant)

1. magento/devdocs#7986: GraphQL: Add company role mutations

### Whatsnew
Added the following mutations in support of B2B companies: [createCompanyRole](https://devdocs.magento.com/guides/v2.4/graphql/mutations/create-company-role.html), [updateCompanyRole](https://devdocs.magento.com/guides/v2.4/graphql/mutations/update-company-role.html), [deleteCompanyRole](https://devdocs.magento.com/guides/v2.4/graphql/mutations/delete-company-role.html), and [isCompanyRoleNameAvailable](https://devdocs.magento.com/guides/v2.4/graphql/queries/is-company-role-name-available.html) query.

### CODE REVIEW

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @andrewbess. Amazing job! Please, check my comments below

**Comment by andrewbess** [comment]
Hello @rogyar 

Thank you for your review.
I completely agree with you.
I have fixed it. Please recheck.
Thank you in advance.

**Comment by andrewbess** [comment]
The merge-conflict has been resolved.

**Review by rogyar — APPROVED** [review]

**Comment by rogyar** [comment]
From the technical perspective looks good. Thanks @andrewbess for your work

**Review by dobooth — CHANGES_REQUESTED** [review]

File: src/guides/v2.4/graphql/mutations/create-company-role.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+      ]
+    }
+  ) {
+    role {
+      id
+      name
+      permissions {
+        id
+        text
+        sort_order
```
[the remark this answers is not in the document]
Comment (dobooth) [reply src/guides/v2.4/graphql/mutations/create-company-role.md:60]: My bad. Incoming code changes it to sort_order

File: src/guides/v2.4/graphql/mutations/update-company-role.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+      name: "Company Admin (updated)"
+    }
+  ) {
+    role {
+      id
+      name
+      permissions {
+        id
+        text
+        sort_order
```
[the remark this answers is not in the document]
Comment (dobooth) [reply src/guides/v2.4/graphql/mutations/update-company-role.md:45]: Nope. This will be correct in the near future. Belay.

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
running tests

**Pull request merged**